staging/vhost-device-sound/src/device.rs

```diff
@@ -322,21 +322,8 @@ impl VhostUserSoundThread {
                         audio_backend
                             .read()
                             .unwrap()
-                            .set_parameters(
-                                stream_id,
-                                ControlMessage {
-                                    kind: code,
-                                    code: VIRTIO_SND_S_OK,
-                                    desc_chain,
-                                    descriptor: desc_hdr,
-                                    vring: vring.clone(),
-                                },
-                            )
+                            .set_parameters(stream_id, request)
                             .unwrap();
```

Comment:
We should remove the unwrap() here and set the appropriate response (`VIRTIO_SND_S_BAD_MSG`, `VIRTIO_SND_S_NOT_SUPP`). If the request returns `Err(Error::UnexpectedAudioBackendConfiguration)` the daemon panics. Let's add a match on possible errors and set the respond code accordingly.

Reply:
Thank you for catching up!